consensus/core/src/authority_service.rs

```diff
@@ -346,6 +346,51 @@ impl<C: CoreThreadDispatcher> NetworkService for AuthorityService<C> {
             .collect();
         Ok((commits, certifier_blocks))
     }
+
+    async fn handle_fetch_latest_blocks(
+        &self,
+        peer: AuthorityIndex,
+        authorities: Vec<AuthorityIndex>,
+    ) -> ConsensusResult<Vec<Bytes>> {
+        fail_point_async!("consensus-rpc-response");
+
+        if authorities.len() > self.context.committee.size() {
+            return Err(ConsensusError::TooManyAuthoritiesProvided(peer));
+        }
+
+        // Ensure that those are valid authorities
+        for authority in &authorities {
+            if !self.context.committee.is_valid_index(*authority) {
+                return Err(ConsensusError::InvalidAuthorityIndex {
+                    index: *authority,
+                    max: self.context.committee.size(),
+                });
+            }
+        }
+
+        // First read from the dag state to find the latest blocks. Then find out if there are any
+        // suspended blocks
+        let mut blocks = vec![];
+        let dag_state = self.dag_state.read();
+        for authority in authorities {
+            let block = dag_state.get_last_block_for_authority(authority);
```

Comment:
Using the DagState only now to figure out the last accepted block for the requested authority. I believe for stronger guarantees we should also look into the `block_manager` for any suspended blocks. For now I've considered this more edge case scenario but let me know if you don't feel the same and I can address on a follow up PR (so avoid this one even bigger)

Reply:
Maybe leave a todo for the suspended block case?